lib/traject/config/folio_config.rb

```diff
@@ -146,18 +146,18 @@ def holdings(record, context)
   end
 end
 
-# TODO: is this the right place to get this from? use instanceRecord or holdingsRecord instead?
-# when are these timestamps updated in FOLIO?
 to_field 'date_cataloged' do |record, accumulator|
-  timestamp = record.record.dig('metadata', 'createdDate')
+  timestamp = record.instance['catalogedDate'] || record.instance.dig('metadata', 'createdDate')
 
-  # when requesting via /source-storage/stream/source-records, you get a unix epoch (e.g. 1658265346880)
+  # when requesting via /source-storage/stream/source-records, you get a unix epoch with milliseconds (e.g. 1658265346880)
   # but when requesting via /source-storage/source-records, you get an ISO8601 string (e.g. "2020-01-01T00:00:00Z")
   # we also need to always convert it to UTC for solr
-  if timestamp.is_a?(Numeric)
-    accumulator << Time.at(timestamp).utc.iso8601
-  elsif timestamp
-    accumulator << Time.iso8601(timestamp).utc.iso8601
+  # division by 1000 is to remove milliseconds from the epoch so it parses correctly
+  case timestamp
+  when Numeric
+    accumulator << Time.at(timestamp / 1000).utc.iso8601
+  when String && (timestamp !~ /NEVER/i)
```

Comment:
Are we populating `catalogedDate` with `NEVER` in FOLIO, or is that a symphony artifact?

Reply:
in slack, @shelleydoljack said:
> coming out of symphony, it’ll be 0. should we leave it that way? i think we’re converting that to NEVER in our export.

so i think we could do it either way — i opted for this for parity with the sirsi config, but maybe there's something simpler.

Reply:
i think if we don't replace 0 with `NEVER`, the current logic will put it in the `Numeric` bucket and it'll become the 1970 date, which seems less accurate than just not indexing a value there.

Reply:
Coming out of Symphony, when date cataloged is never, the API returns 0. When migrating to FOLIO, this should translate to what FOLIO does, which is not write a value to the instance field catalogedDate. So, for FOLIO, line 150 and 158 matter and Symphony-migrated records in FOLIO won't need line 159. However, because IndexData is taking date cataloged from MARC 008, which could be a string (e.g. "19uu-uu-uu", but that's a migration error, they should not write to that field when data looks like this), then we should account for that in some way, like not adding it to the `date_cataloged` solr field.